src/main/java/com/linkedin/kmf/services/ProduceService.java

```diff
@@ -150,7 +150,7 @@ private void initializeProducer() throws Exception {
   public void start() {
     if (_running.compareAndSet(false, true)) {
       initializeStateForPartitions();
-      _handleNewPartitionsExecutor.scheduleWithFixedDelay(new NewPartitionHandler(), 40_000, 40_000, TimeUnit.MILLISECONDS);
+      _handleNewPartitionsExecutor.scheduleWithFixedDelay(new NewPartitionHandler(), 1000, 30000, TimeUnit.MILLISECONDS);
```

Comment:
By having a longer wait time it's more likely that the topic management service will have time to create the topic.   That means fewer questions I need to answer from the SREs when they start the monitors and see something strange.  If you think this is not going to happen often enough then I will defer to your greater experience in this matter.